omp/CMakeLists.txt

```diff
@@ -27,8 +27,7 @@ target_sources(ginkgo_omp
         stop/residual_norm_reduction_kernels.cpp)
 
 ginkgo_compile_features(ginkgo_omp)
-target_link_libraries(ginkgo_omp PRIVATE "${OpenMP_CXX_LIBRARIES}")
-target_compile_options(ginkgo_omp PRIVATE "${OpenMP_CXX_FLAGS}")
+target_link_libraries(ginkgo_omp PRIVATE OpenMP::OpenMP_CXX)
```

Comment:
Since we are already cleaning this up, might make sense to get rid of the other uses of these variables (below and in GinkgoConfig.cmake.in). @tcojean can we safely remove this from the installation config?

Reply:
As far as I know, no. I actually want this to stay the same. There were multiple bugs previously which made the `OpenMP::OpenMP_whatever` targets provide no flag on some compilers, which is why I went with this verison.

Reply:
I will keep the flags. But instead of the `${OpenMP_CXX_LIBRARIES}`, I use the targets. It seems to work on the Gitlab CI as well.